CandyMarket/CandyMarket/DataAccess/CandyRepo.cs

```diff
@@ -181,6 +197,33 @@ from UserCandies
             }
         }
 
+        // ConsumeRandomCandy: finds UserCandyId for oldest piece of a random candy by flavor and calls EatCandy
+        public UserCandyDetailed ConsumeRandomCandy(string flavor, int userId)
+        {
+            var sql = @"select top 1 with ties UserCandyId
+                        from UserCandies uc
+                            join Candies c
+                            on uc.candyId = c.candyId
+                        where isConsumed = 0
+                        and flavorCategory = @Flavor
+                        and userId = @UserId
+                        order by dateReceived";
```

Comment:
Add desc after date received to ensure dates are ordered oldest to newest

Reply:
using desc puts the most recent dates first.  It seemed counter-intuitive to me at first too

Reply:
So weird...